src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1059,13 +997,12 @@ GraphPatternOperation Visitor::visit(Parser::ServiceGraphPatternContext* ctx) {
   //
   // TODO: Also support variables. The semantics is to make a connection for
   // each IRI matching the variable and take the union of the results.
-  GraphTerm varOrIri = visit(ctx->varOrIri());
+  VarOrIri varOrIri = visit(ctx->varOrIri());
   if (std::holds_alternative<Variable>(varOrIri)) {
     reportNotSupported(ctx->varOrIri(), "Variable endpoint in SERVICE is");
   }
-  AD_CONTRACT_CHECK(std::holds_alternative<Iri>(varOrIri));
-  auto serviceIri =
-      TripleComponent::Iri::fromIriref(std::get<Iri>(varOrIri).iri());
+  AD_CONTRACT_CHECK(std::holds_alternative<TripleComponent::Iri>(varOrIri));
+  auto serviceIri = std::get<TripleComponent::Iri>(varOrIri);
```

Comment:
This `if/else` can now also be `std::visit()` 
with a lambda that explicitly returns the `Iri`, so auto serviceIri = std::visit(..., ...)`.